lib/modules/newCommentCount.js

```diff
@@ -389,7 +389,11 @@ async function cleanCache() {
 	for (const i in commentCounts) {
 		if (commentCounts[i] && ((now - commentCounts[i].updateTime) > (86400000 * module.options.cleanComments.value))) {
 			// Do not automatically delete comments belonging to actively subscribed threads
-			if (skip && typeof commentCounts[i].subscriptionDate !== 'undefined') {
+			if (skip) {
+				// Delete expired subscriptions
+				if (typeof commentCounts[i].subscriptionDate === 'undefined') {
```

Comment:
subscriptions should be cleaned per the subscription length option, comment threads that aren't subscribed should be cleared per the clean comments option. More specifically it looks like Erik has the good ideas.